src/storage.h

```diff
@@ -0,0 +1,253 @@
+/// residence_time is the minimum number of timesteps between receiving and offering
+/// in_recipe (optional) describes the incoming resource by recipe
+/// 
+/// @section optionalparams Optional Parameters
+/// max_inv_size is the maximum capacity of the inventory storage
+/// throughput is the maximum processing capacity per timestep
+///
+/// @section detailed Detailed Behavior
+/// 
+/// Tick:
+/// Nothing really happens on the tick. 
+///
+/// Tock:
+/// On the tock, any material that has been waiting for long enough (delay 
+/// time) is placed in the stocks buffer.
+///
+/// Any brand new inventory that was received in this timestep is placed into 
+/// the processing queue to begin waiting. 
+/// 
+/// Making Requests:
+/// This facility requests all of the in_commod that it can.
+///
+/// Receiving Resources:
+/// Anything of the in_commod that is received by this facility goes into the 
+/// inventory.
+///
+/// Making Offers:
+/// Any stocks material in the stocks buffer is offered to the market.
+///
+/// Sending Resources:
+/// Matched resources are sent immediately.
+class Storage 
+  : public cyclus::Facility,
+    public cyclus::toolkit::CommodityProducer {
+ public:  
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  Storage(cyclus::Context* ctx);
+  
+  #pragma cyclus decl
+
+  #pragma cyclus note {"doc": "Storage is a simple facility which accepts any number of commodities " \
+                              "and holds them for a user specified amount of time. The commodities accepted "\
+                              "are chosen based on the specified preferences list. Once the desired amount of material "\
+                              "has entered the facility it is passed into a 'processing' buffer where it is held until "\
+                              "the residence time has passed. The material is then passed into a 'ready' buffer where it is "\
+                              "queued for removal. Currently, all input commodities are lumped into a single output commodity. "\
+                              "Storage also has the functionality to handle materials in discrete or continuous batches. Discrete "\
+                              "mode, which is the default, does not split or combine material batches. Continuous mode, however, "\
+                              "divides material batches if necessary in order to push materials through the facility as quickly "\
+                              "as possible."}
+
+  /// A verbose printer for the Storage Facility
+  virtual std::string str();
+
+  // --- Facility Members ---
+  
+  // --- Agent Members ---
+  /// Sets up the Storage Facility's trade requests
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Storage.
+  virtual void Tick();
+
+  /// The handleTick function specific to the Storage.
+  virtual void Tock();
+
+  /* --- Storage Members --- */
```

Comment:
Since we don't have a good picture of how archetype extension will be done in practice, these sort of getters/setters aren't necessary now - even if they may be in the future.  So I've generally leaned toward not creating any.